server/src/main/java/keywhiz/service/daos/GroupDAO.java

```diff
@@ -93,22 +118,57 @@ public void deleteGroup(Group group) {
           .set(SECRETS.OWNER, (Long) null)
           .where(SECRETS.OWNER.eq(group.getId()))
           .execute();
+      DSL.using(configuration)
+          .update(GROUPS)
+          .set(GROUPS.OWNER, (Long) null)
+          .where(GROUPS.OWNER.eq(group.getId()))
+          .execute();
     });
   }
 
   public Optional<Group> getGroup(String name) {
     GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.NAME.eq(name));
-    return Optional.ofNullable(r).map(groupMapper::map);
+    return Optional.ofNullable(toGroup(r));
```

Comment:
nit: Is there a way to get the group and owner group in one query?

Reply:
After poking around at this for a couple hours, I believe I have misjudged Jooq as a result of an earlier programming error (it only falls back to string matching when proper aliasing is missing). It turns out that it's trivial but non-obvious to correctly perform a self-join. The new code is much cleaner and also efficient in that we're only fetching the name of the owner since that's the only field we care about. It also made me realize two things:

1. There are a ton of places in Keywhiz were we hydrate entire objects only to throw away every field but one (typically, the secret name). We could do so much better there.
2. I need to go back and re-implement the owner fetching for secrets.

Reply:
Awesome @mmontgomery-square! Thanks for looking into this.